app/models/billing/page_unlock_promo_code.rb

```diff
@@ -0,0 +1,32 @@
+class PageUnlockPromoCode < ApplicationRecord
+  has_many :promotions
+  has_many :users, -> { distinct }, through: :promotions
+
+  def unlocked_page_types
+    page_types.split('|') & Rails.application.config.content_types[:all].map(&:name)
```

Comment:
Is page_types a dB column? Why not use something like https://www.altcademy.com/questions/store-arrays-or-hashes-in-rails-activerecord so you don’t have to split it?

Reply:
Ah, interesting. I tend not to stray into a lot of db-specific types because I use sqlite locally and postgres in production, but that looks just abstracted enough to fit perfectly here. :+1:

Reply:
Luckily this is built into activerecord and serializes to a string in the db, so the backend doesn't need to be specific.

Reply:
Seems to be working perfectly. Thanks for the suggestion. :snowboarder: